localstack/services/secretsmanager/provider.py

```diff
@@ -297,10 +369,37 @@ def update_secret(
         self, context: RequestContext, request: UpdateSecretRequest
     ) -> UpdateSecretResponse:
         # if we're modifying the value of the secret, ClientRequestToken is required
-        if any(key for key in request if key in ("SecretBinary", "SecretString")):
-            self._raise_if_missing_client_req_token(request)
-        self._raise_if_invalid_secret_id(request["SecretId"])
-        return call_moto(context, request)
+        secret_id = request["SecretId"]
+        secret_string = request.get("SecretString")
+        secret_binary = request.get("SecretBinary")
+        description = request.get("Description")
+        kms_key_id = request.get("KmsKeyId")
+        client_req_token = request.get("ClientRequestToken")
+        self._raise_if_invalid_secret_id(secret_id)
+        self._raise_if_missing_client_req_token(request)
+
+        backend = SecretsmanagerProvider.get_moto_backend_for_resource(secret_id, context)
+        try:
+            secret = backend.update_secret(
+                secret_id,
+                description=description,
+                secret_string=secret_string,
+                secret_binary=secret_binary,
+                client_request_token=client_req_token,
+                kms_key_id=kms_key_id,
+            )
+        except MotoSecretNotFoundException:
+            raise ResourceNotFoundException("Secrets Manager can't find the specified secret.")
+        except MotoOperationNotPermittedOnReplica:
+            raise InvalidRequestException(
+                "Operation not permitted on a replica secret. Call must be made in primary secret's region."
+            )
+        except MotoInvalidRequestException:
+            raise InvalidRequestException(
+                "An error occurred (InvalidRequestException) when calling the UpdateSecret operation: "
+                "You can't perform this operation on the secret because it was marked for deletion."
+            )
+        return UpdateSecretResponse(**json.loads(secret))
```

Comment:
Could you confirm that the responses for this (and other modified) operations are AWS validated at least once in the tests?